src/test/regress/expected/pg16.out

```diff
@@ -65,6 +65,214 @@ SET citus.log_remote_commands TO OFF;
+
+(1 row)
+
+INSERT INTO test_is_json(js) VALUES
+ (NULL),
+ (''),
+ ('123'),
+ ('"aaa "'),
+ ('true'),
+ ('null'),
+ ('[]'),
+ ('[1, "2", {}]'),
+ ('{}'),
+ ('{ "a": 1, "b": null }'),
+ ('{ "a": 1, "a": null }'),
+ ('{ "a": 1, "b": [{ "a": 1 }, { "a": 2 }] }'),
+ ('{ "a": 1, "b": [{ "a": 1, "b": 0, "a": 2 }] }'),
+ ('aaa'),
+ ('{a:1}'),
+ ('["a",]');
+-- run IS JSON predicate in the worker nodes
+SELECT
+	js,
+	js IS JSON "JSON",
+	js IS NOT JSON "NOT JSON",
+	js IS JSON VALUE "VALUE",
+	js IS JSON OBJECT "OBJECT",
+	js IS JSON ARRAY "ARRAY",
+	js IS JSON SCALAR "SCALAR",
+	js IS JSON WITHOUT UNIQUE KEYS "WITHOUT UNIQUE",
+	js IS JSON WITH UNIQUE KEYS "WITH UNIQUE"
+FROM
+	test_is_json ORDER BY js;
+                      js                       | JSON | NOT JSON | VALUE | OBJECT | ARRAY | SCALAR | WITHOUT UNIQUE | WITH UNIQUE
+---------------------------------------------------------------------
+                                               | f    | t        | f     | f      | f     | f      | f              | f
+ "aaa "                                        | t    | f        | t     | f      | f     | t      | t              | t
+ 123                                           | t    | f        | t     | f      | f     | t      | t              | t
+ ["a",]                                        | f    | t        | f     | f      | f     | f      | f              | f
+ [1, "2", {}]                                  | t    | f        | t     | f      | t     | f      | t              | t
+ []                                            | t    | f        | t     | f      | t     | f      | t              | t
+ aaa                                           | f    | t        | f     | f      | f     | f      | f              | f
+ null                                          | t    | f        | t     | f      | f     | t      | t              | t
+ true                                          | t    | f        | t     | f      | f     | t      | t              | t
+ { "a": 1, "a": null }                         | t    | f        | t     | t      | f     | f      | t              | f
+ { "a": 1, "b": [{ "a": 1 }, { "a": 2 }] }     | t    | f        | t     | t      | f     | f      | t              | t
+ { "a": 1, "b": [{ "a": 1, "b": 0, "a": 2 }] } | t    | f        | t     | t      | f     | f      | t              | f
+ { "a": 1, "b": null }                         | t    | f        | t     | t      | f     | f      | t              | t
+ {a:1}                                         | f    | t        | f     | f      | f     | f      | f              | f
+ {}                                            | t    | f        | t     | t      | f     | f      | t              | t
+                                               |      |          |       |        |       |        |                |
+(16 rows)
+
+-- pull the data, and run IS JSON predicate in the coordinator
+WITH pulled_data as (SELECT js FROM test_is_json)
```

Comment:
We might to cte_inlining, so maybe add WITH pulled_data as (SELECT js FROM test_is_json **OFFSET 0**) to make sure we always pull to the coordinator